lib/is_external_link.js

```diff
@@ -13,11 +13,21 @@ const cache = new Cache();
 
 function isExternalLink(input, sitehost, exclude) {
   return cache.apply(`${input}-${sitehost}-${exclude}`, () => {
+    // Return false early for internal link
+    if (!/^(\/\/|http(s)?:)/.test(input)) return false;
+
     sitehost = parse(sitehost).hostname || sitehost;
 
     if (!sitehost) return false;
-    // handle relative url
-    const data = new URL(input, `http://${sitehost}`);
+
+    // handle relative url and invalid url
+    let data;
+    try {
+      data = new URL(input, `http://${sitehost}`);
+    } catch (e) { }
```

Comment:
How about `hexo.log.warn(...)` to alert the user that there has an invalid URL?

Reply:
Since hexo-util has no `hexo-log`'s dependency, there is no way we can use `hexo.log.warn` in hexo-util.